### ISSUE TITLE
Doesn't refresh relationship when updating the foreign_key of a model.

### ISSUE THREAD

**Issue by jazerix** [laravel/framework#25803]
- Laravel Version: 5.7.6
- PHP Version: 7.2.9
- Database Driver & Version: MySQL 5.7

### Description:
If you change the foreign key on a model that has the relationship already loaded, the model won't refresh the relationship and will therefore return the different relation when called later on. 

I'm not entirely sure this is actually an issue, but I stumbled upon this while doing some integration tests. Once the 

### Steps To Reproduce:

User model:
`public function phone()
    {
        return $this->belongsTo(phone::class, 'phone_id');
    }`

testBasicTest:
       `

        $phoneId = factory(\App\Phone::class)->create()->id;
        $user = factory(\App\User::class)->create([              
            'phone_id' => $phoneId
        ]);
        
        $user->phone->id; //remove this

        $newPhone = factory(\App\Phone::class)->create();
        $user->phone_id = $newPhone->id;
        $user->save();

        $this->assertEquals($newPhone->id, $user->phone()->first()->id); //works
        $this->assertEquals($newPhone->id, $user->phone->id); //doesn't work`

If you remove the `$user->phone->id;` line this test will pass, otherwise it wont.

**Comment by mfn** [comment]
I believe this is simply intentional: Not everyone want or can reload relations implicitly due to this and there may be unexpected edge cases (e.g. you set a relation you haven't yet persisted, then change the FK).

The relation and FK property are separate things. The latter is implicitly (or through configuration) used by the former. But otherwise, the relation is independent.

You _can_ use something like `->refresh()` on the model:
> Reload the current model instance with fresh attributes from the database.

Or manually call `->load(…)`.

 TL;DR: IMHO by design as there are multiple solutions and the decision is to don't make a decision and leave it to the dev.

**Comment by laurencei** [comment]
Thanks @mfn.

**Comment by jazerix** [comment]
I think you're right! While going through the source code I wasn't able to produce any fix for this, without know the relations of the model beforehand. 

I'm gonna go with associate instead, thanks! :)
